compiler/rustc_codegen_ssa/src/mir/analyze.rs

```diff
@@ -286,7 +286,18 @@ impl<'mir, 'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> Visitor<'tcx>
                     Some(assignment_location) => {
                         assignment_location.dominates(location, &self.dominators)
                     }
-                    None => false,
+                    None => {
+                        debug!("No first assignment found for {:?}", local);
+                        // We have not seen any assignment to the local yet,
+                        // but before marking not_ssa, check if it is a ZST,
+                        // in which case we don't need to initialize the local.
+                        let ty = self.fx.mir.local_decls[local].ty;
+                        let ty = self.fx.monomorphize(ty);
+
+                        let is_zst = self.fx.cx.layout_of(ty).is_zst();
+                        debug!("is_zst: {}", is_zst);
+                        is_zst
```

Comment:
I wonder if anything breaks if we _do_ in fact just go ahead and unconditionally mark any variable without assignments as "ssa".

Reply:
I tried replacing `is_zst` with `true` which at least didn't break test/{mir-opt,codegen}. Should i push that instead of the current zst handling?

Reply:
Lets not do that in the current PR, especially because apparently this was added as part of a ICE fix, but without tests. I will experiment with it separately.